src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.libraries;
+
+import com.google.cloud.tools.libraries.json.CloudLibrary;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/** Returns helpful metadata for supported Google Cloud libraries. */
+public final class CloudLibraries {
+
+  private static final String LIBRARIES_JSON = "libraries.json";
+
+  private final String librariesJsonPath;
+
+  @VisibleForTesting
+  CloudLibraries(String librariesJsonPath) {
+    this.librariesJsonPath = librariesJsonPath;
+  }
+
+  /**
+   * Returns the list of {@link CloudLibrary} objects deserialized from the {@code libraries.json}
+   * file.
+   *
+   * @throws IOException if there was a problem reading the {@code libraries.json} file
```

Comment:
Since this is bundled in the jar this is a case where a runtime exception is more appropriate.

Reply:
Could you explain more? IMO throwing an unchecked exception here does not give the caller an opportunity to gracefully degrade their user experience if this happens to fail (which could happen even if `libraries.json` does exist in the JAR). Sure they could catch any `Throwable` or `RuntimeException` coming from this method, but the context isn't as clear and it's not enforced by the method signature.

Reply:
Checked exceptions are for problems external to the program such as an IOException or syntax error in an external document.
Runtime exceptions are for program bugs.

This is a really unusual case where the JSON is not an external file but program source code bundled into and read from the jar. A failure here is a bug in the program. Thus runtime exception. 

Most similar cases should indeed be checked exceptions.

Reply:
@nkibler does have a point that a failure here doesn't always indicate a bug.  It could be an IO issue of some sort.